catalog/async/catalog-async-processingframework/src/main/java/org/codice/ddf/catalog/async/processingframework/impl/InMemoryProcessingFramework.java

```diff
@@ -0,0 +1,326 @@
+        if (CollectionUtils.isNotEmpty(contentItemsToUpdate)) {
+            LOGGER.trace("Storing content item updates(s)");
+
+            UpdateStorageRequest updateStorageRequest =
+                    new UpdateStorageRequestImpl(new ArrayList<>(contentItemsToUpdate), properties);
+
+            Subject subject = (Subject) updateStorageRequest.getProperties()
+                    .get(SecurityConstants.SECURITY_SUBJECT);
+            if (subject == null) {
+                LOGGER.debug(
+                        "No subject to send UpdateStorageRequest. Updates will not be sent back to the catalog");
+            } else {
+                subject.execute(() -> {
+                    try {
+                        catalogFramework.update(updateStorageRequest);
+                        LOGGER.debug("Successfully completed update storage request");
+                    } catch (IngestException | SourceUnavailableException e) {
+                        LOGGER.debug("Unable to complete update storage request", e);
+                    }
+
+                    return null;
+                });
+            }
+        } else {
+            LOGGER.debug("No content items to update");
+        }
+    }
+
+    private void close(List<TemporaryFileBackedOutputStream> tfbosToCleanUp) {
+        tfbosToCleanUp.forEach(tfbos -> {
+            try {
+                tfbos.close();
+            } catch (IOException e) {
+                LOGGER.debug("Failed to cleanup temporary file.");
+            }
+        });
+    }
+
+    private void storeMetacardUpdates(Set<Metacard> metacardsToUpdate,
+            Map<String, Serializable> properties) {
+        if (CollectionUtils.isNotEmpty(metacardsToUpdate)) {
+            LOGGER.trace("Storing metacard updates");
+
+            List<Map.Entry<Serializable, Metacard>> updateList = metacardsToUpdate.stream()
+                    .map(metacard -> new AbstractMap.SimpleEntry<Serializable, Metacard>(metacard.getId(),
+                            metacard))
+                    .collect(Collectors.toList());
+
+            UpdateRequest updateMetacardsRequest = new UpdateRequestImpl(updateList,
+                    UpdateRequest.UPDATE_BY_ID,
+                    properties);
+
+            Subject subject = (Subject) updateMetacardsRequest.getProperties()
+                    .get(SecurityConstants.SECURITY_SUBJECT);
+
+            if (subject == null) {
+                LOGGER.debug(
+                        "No subject to send UpdateRequest. Updates will not be sent back to the catalog.");
+            } else {
+                subject.execute(() -> {
+                    try {
+                        catalogFramework.update(updateMetacardsRequest);
+                        LOGGER.debug("Successfully completed update metacards request");
+                    } catch (IngestException | SourceUnavailableException e) {
+                        LOGGER.debug("Unable to complete update request", e);
```

Comment:
I think DEBUG is the most appropriate based on https://codice.atlassian.net/wiki/display/DDF/Log+Levels. The only other level that may appropriate is WARN, but I'm not sure there is any admin action to take here if this occurs, or if we can give them any steps to correct this.